src/openrct2/scenario/Scenario.cpp

```diff
@@ -415,60 +416,69 @@ void scenario_update()
     }
     scenario_update_daynight_cycle();
 }
-
 /**
  *
  *  rct2: 0x006744A9
  */
 static int32_t scenario_create_ducks()
 {
-    int32_t i, j, r, c, x, y, waterZ, centreWaterZ, x2, y2;
+    CoordsXY centrePos;
+    centrePos.x = 64 + (scenario_rand_max(MAXIMUM_MAP_SIZE_PRACTICAL) * 32);
```

Comment:
It looks like this shifts the area from being in the centre and skipping the outer ring of invisible tiles, to one tile in both x and y directions. AFAICS, this should be changed to `32 + ...`.

Reply:
You have to consider that it subtracts 64 some lines below so that would go negative